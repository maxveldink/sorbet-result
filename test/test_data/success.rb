# typed: true
# frozen_string_literal: true

class TestGenerics
  extend T::Sig

  sig { params(should_succeed: T::Boolean).returns(Typed::Result[Integer, String]) }
  def test_generic_initializer(should_succeed)
    if should_succeed
      Typed::Success.new("1")
    else
      Typed::Failure.new("")
    end
  end

  sig { returns(Typed::Success[String]) }
  def test_inferred_payload_type
    success = Typed::Success.new("success")

    T.assert_type!(success.payload, String)
    success
  end

  sig { returns(Typed::Success[Integer]) }
  def test_explicit_payload_type
    Typed::Success[Integer].new("success")
  end

  sig { void }
  def test_payload_or
    success = Typed::Success.new("success")

    success.payload_or(1).upcase
  end
end
