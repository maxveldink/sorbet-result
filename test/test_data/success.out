test/test_data/success.rb:14: Expected `Typed::Result[Integer, String]` but found `T.any(Typed::Success[String], Typed::Failure[String])` for method result type https://srb.help/7005
    14 |  end
          ^^^
  Expected `Typed::Result[Integer, String]` for result type of method `test_generic_initializer`:
    test/test_data/success.rb:8:
     8 |  def test_generic_initializer(should_succeed)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Got `T.any(Typed::Success[String], Typed::Failure[String])` originating from:
    test/test_data/success.rb:10:
    10 |      Typed::Success.new("1")
              ^^^^^^^^^^^^^^^^^^^^^^^
    test/test_data/success.rb:12:
    12 |      Typed::Failure.new("")
              ^^^^^^^^^^^^^^^^^^^^^^

test/test_data/success.rb:26: Expected `Integer` but found `String("success")` for argument `payload` https://srb.help/7002
    26 |    Typed::Success[Integer].new("success")
                                        ^^^^^^^^^
  Expected `Integer` for argument `payload` of method `Typed::Success#initialize`:
    ./lib/typed/success.rb:30:
    30 |    sig { params(payload: Payload).void }
                         ^^^^^^^
  Got `String("success")` originating from:
    test/test_data/success.rb:26:
    26 |    Typed::Success[Integer].new("success")
                                        ^^^^^^^^^

test/test_data/success.rb:33: Method `upcase` does not exist on `Integer` component of `T.any(String, Integer)` https://srb.help/7003
    33 |    success.payload_or(1).upcase
                                  ^^^^^^
  Got `T.any(String, Integer)` originating from:
    test/test_data/success.rb:33:
    33 |    success.payload_or(1).upcase
            ^^^^^^^^^^^^^^^^^^^^^
  Did you mean `phase`? Use `-a` to autocorrect
    test/test_data/success.rb:33: Replace with `phase`
    33 |    success.payload_or(1).upcase
                                  ^^^^^^
    https://github.com/sorbet/sorbet/tree/6cfdcc56f3b81d8024b6423167d745f4bfd92258/rbi/core/integer.rbi#L921: Defined here
     921 |  def phase(); end
            ^^^^^^^^^^^
